src/systemctl/systemctl-list-units.c

```diff
@@ -369,38 +369,36 @@ static int output_sockets_list(struct socket_info *socket_infos, unsigned cs) {
 
         (void) table_set_empty_string(table, "-");
 
-        if (cs) {
-                for (struct socket_info *s = socket_infos; s < socket_infos + cs; s++) {
```

Comment:
I have the suspicion some of the static checkers are not going to like this, given that socket_infos might be NULL here. i.e. the static checkers really don't like it if one days maths on NULL pointers, even if the value you add to them is 0 itself...

Hence, change the second part of the for expression from this:

```c
…; s < socket_infos + cs; …
```

to

```c
…; socket_infos && (s < socket_infos + cs); …
```

then the static checkers shouldn't mind